src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/FlagAuthoritySystem.java

```diff
@@ -113,4 +123,59 @@ public void onGiveItemToCharacterHoldItem(GiveItemEvent event, EntityRef item, I
                 delayManager.cancelDelayedAction(item, LASUtils.DROPPED_FLAG);
         }
     }
+    /**
+     * Iterates through the inventory to locate the flag and then sends a {@link DropItemRequest} to remove the flag from
+     * the player's inventory.
+     *
+     * @param event The FlagDrop event sent to the player.
+     * @param targetPlayer The entity which should drop the flag.
+     */
+    @ReceiveEvent
+    public void dropFlagRequest(DropFlagEvent event, EntityRef targetPlayer) {
+        int inventorySize = inventoryManager.getNumSlots(targetPlayer);
+        for (int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
+            EntityRef inventorySlot = inventoryManager.getItemInSlot(targetPlayer, slotNumber);
+            Optional<FlagComponent> flagComponent = Optional.ofNullable(inventoryManager.getItemInSlot(targetPlayer,
+                    slotNumber).getComponent(FlagComponent.class));
+            flagComponent.ifPresent(flag -> {
+                if (flag.team.equals(LASUtils.getOppositionTeam(targetPlayer.getComponent(LASTeamComponent.class).team))) {
+                    flagSlot = inventorySlot;
+                }
+            });
+        }
+        Vector3fc startPosition = targetPlayer.getComponent(LocationComponent.class).getLocalPosition();
+        Vector3f impulse = event.getAttackingPlayer().getComponent(LocationComponent.class).getLocalPosition()
```

Comment:
ah, I see we need the attacking player to calculate the impulse... seems a bit weird but oh well...